File: src/test/java/no/nav/familie/ks/oppslag/tilgangskontroll/EgenAnsattTestConfig.java
package no.nav.familie.ks.oppslag.tilgangskontroll;

import no.nav.familie.ks.oppslag.egenansatt.internal.EgenAnsattConsumer;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Primary;
import org.springframework.context.annotation.Profile;

import static org.mockito.Mockito.doNothing;
import static org.mockito.Mockito.mock;

@Configuration
public class EgenAnsattTestConfig {

    @Bean
    @Profile("mock-egenansatt")
    @Primary
    public EgenAnsattConsumer egenAnsattConsumerMock() {
        EgenAnsattConsumer egenAnsattConsumer = mock(EgenAnsattConsumer.class);

        doNothing().when(egenAnsattConsumer).ping();
        return egenAnsattConsumer;
    }
}

File: src/test/java/no/nav/familie/integrasjoner/tilgangskontroll/EgenAnsattTestConfig.kt
package no.nav.familie.integrasjoner.tilgangskontroll

import io.mockk.mockk
import no.nav.familie.integrasjoner.client.soap.EgenAnsattSoapClient
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Primary
import org.springframework.context.annotation.Profile

@Configuration
class EgenAnsattTestConfig {

    @Bean
    @Primary
    @Profile("mock-egenansatt")
    fun mockEgenAnsattSoapClient(): EgenAnsattSoapClient {
        return mockk(relaxed = true)
    }
}
